src/test/ui/rfc-2632-const-trait-impl/specialization/const-default-non-const-specialized.rs

```diff
@@ -0,0 +1,37 @@
+// Tests that a const default trait impl can be specialized by a non-const trait
+// impl, but that the specializing impl cannot be used in a const context.
```

Comment:
alright :thinking: so this is intended, but it feels wrong to me

taking this approach ends up with errors during monomorphization which don't get detected with `cargo check` and tend to have back error locations. so these should be avoided

Reply:
So I suppose you disagree with the premise of https://github.com/rust-lang/rust/pull/95292, then. I think I understand where you're coming from. FWIW, though, the error experience seems acceptable to me in the example of this test.

I'll have to test that disallowing this still enables the [result tracing](https://github.com/BGR360/rust/pull/1) functionality to work. That's the only reason I'm trying to push this change through. If it doesn't change things, then I'd be willing to disallow it. I would like to better understand some of the things you're saying though rather than just take your word for it.

Reply:
answered more generally in https://github.com/rust-lang/rust/pull/95292#discussion_r969052397

> So I suppose you disagree with the premise of https://github.com/rust-lang/rust/pull/95292, then.

did you mean to mention #95292 (this PR) here?

Reply:
> did you mean to mention #95292 (this PR) here?

Nope, lol, I meant to link https://github.com/rust-lang/rust/issues/95187, one of the two issues this PR aimed to fix